### PR TITLE
Add jump and ouch sounds for Doom Legacy

### PR DESCRIPTION
**Pull request by Mr795**
I was just randomly thinking about this when i looked inside of the doom legacy wad and seen the sounds "DSJUMP" and "DSOUCH" then i just thought about making a PR for this, the jump sound is just a reversed version of DSPLPAIN and the ouch sound is just DSPLPAIN. (i'm wondering if this'll cause problems with vanilla compatibility or not)

### CODE REVIEW

**Comment by mc776** [comment]
the DSJUMP sounds a lot less bad than I thought it would be.

Is it confirmed that Doom Legacy actually uses these though? I haven't used that port in 20 years.

**Comment by mc776** [comment]
If we add these, I think BUILDCFG should be updated to add right after line 326:
```
; Additional player sounds used in Doom Legacy
DSOUCH	= dsplpain
DSJUMP
```
with no separate actual dsouch.wav file.

**Review by mc776 — CHANGES_REQUESTED** [review]
needs buildcfg update; also since we're just reusing the existing pain sound we might as well link to it in buildcfg instead of reduplicating the file.

**Comment by mc776** [comment]
I'm looking at legacy.wad and there is a *shitton* of stuff that's extremely specific to id in there. It would require a *huge* amount of stuff to get everything up to speed that bloats the download for the sake of what is now a relatively obscure sourceport.

I'm not sure we should proceed with this.

**Comment by Mr795** [comment]
> the DSJUMP sounds a lot less bad than I thought it would be.
> 
> Is it confirmed that Doom Legacy actually uses these though? I haven't used that port in 20 years.

i've used it before and i do confirm doom legacy uses it, this is optional but ZDoom uses it as well if its in a wad.

**Comment by Mr795** [comment]
> If we add these, I think BUILDCFG should be updated to add right after line 326:
> 
> ```
> ; Additional player sounds used in Doom Legacy
> DSOUCH	= dsplpain
> DSJUMP
> ```
> 
> with no separate actual dsouch.wav file.

after or on 326?

**Comment by Mr795** [comment]
i think i might need to fix how the sound plays in the wad, christ it was put in a microwave.

**Comment by mc776** [comment]
Had to look up where DSOUCH is used:
http://www.doomlegends.com/emporium/tutorials/specs03.html

player hitting head against ceiling... maybe the usual pain sound isn't that useful, how about an impact like the new DSOOF?

File: buildcfg.txt [hunk 1]
```diff
@@ -436,6 +436,10 @@ DPSKESIT
 DPSKEATK
 DPRADIO
 
+; Additional player sounds used in Doom Legacy
```
Comment (mc776) [inline buildcfg.txt:439]: this needs to be outside of the "if Phase 2" block.

**Comment by Mr795** [comment]
> Had to look up where DSOUCH is used: http://www.doomlegends.com/emporium/tutorials/specs03.html
> 
> player hitting head against ceiling... maybe the usual pain sound isn't that useful, how about an impact like the new DSOOF?

i think that just gave me an idea.

**Comment by Mr795** [comment]
> > Had to look up where DSOUCH is used: http://www.doomlegends.com/emporium/tutorials/specs03.html
> > player hitting head against ceiling... maybe the usual pain sound isn't that useful, how about an impact like the new DSOOF?
> 
> i think that just gave me an idea.

so i mixed dspunch and dsplpain to make a separate dsouch sound

**Comment by mc776** [comment]
I notice the sounds were uploaded in 44100, I've downsampled them to 22050 here:
[grunting.zip](https://github.com/freedoom/freedoom/files/13852551/grunting.zip)

File: buildcfg.txt [hunk 2]
```diff
@@ -324,6 +324,10 @@ DSSKEATK
 DSRADIO
 #endif /* #ifdef PHASE2 */
 
+; Additional player sounds used in Doom Legacy
+DSOUCH
```
Comment (mc776) [inline buildcfg.txt:328]:
```
; Additional player sounds used in Doom Legacy and ZDoom
DSOUCH ; hit head on ceiling (Doom Legacy only)
```

**Review by mc776 — APPROVED** [review]
Approving this for now but let's see if anyone else has any input.

**Comment by mc776** [comment]
Given the discovery of #1261 I take it people have been trying this out already, so I might as well merge this.

**Pull request merged**